### PR TITLE
Cleanup rolling-upgrade support code

### PR DESCRIPTION
**Pull request by vbekiaris**
Cleanup 4.0 -> 4.1 rolling upgrade support code.

First commit addresses various instances of RU related code. Changes in the second commit address removal of deprecated methods in partitioning & migration packages.

EE-side counterpart: https://github.com/hazelcast/hazelcast-enterprise/pull/3875

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/internal/partition/impl/PartitionStateManager.java [hunk 1]
```diff
@@ -464,18 +428,13 @@ int replaceMember(Member oldMember, Member newMember) {
...
-            //RU_COMPAT_4_0
-            version = getVersion();
-        }
-        return new PartitionTableView(getPartitionsCopy(true), version);
+        return new PartitionTableView(getPartitionsCopy(true));
     }
 
     @Override
     public void onClusterVersionChange(Version newVersion) {
         if (newVersion.isEqualTo(Versions.V4_1) && initialized) {
```
Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/internal/partition/impl/PartitionStateManager.java:436]: could this version comparison be removed  also?--> `newVersion.isEqualTo(Versions.V4_1)`

**Review by ahmetmircik — APPROVED** [review]
1 minor comment + LGTM

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/3129/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/3129/Hazelcast-pr-builder-3129.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] COMPILATION ERROR : 
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/internal/memory/impl/UnsafeUtil.java:[22,15] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:2.5.1:compile (default-compile) on project hazelcast: Compilation failure: Compilation failure: 
--------------------------
---------ERRORS-----------
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/internal/memory/impl/UnsafeUtil.java:[22,15] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/internal/memory/impl/UnsafeUtil.java:[22,15] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/internal/memory/impl/UnsafeUtil.java:[22,15] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/internal/partition/impl/InternalPartitionServiceImpl.java:[431,37] error: cannot find symbol
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/internal/memory/impl/UnsafeUtil.java:[49,8] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/internal/memory/impl/UnsafeUtil.java:[67,19] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/internal/memory/impl/UnsafeUtil.java:[69,19] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/internal/memory/impl/UnsafeUtil.java:[71,70] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/internal/memory/impl/UnsafeUtil.java:[73,23] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/internal/memory/impl/UnsafeUtil.java:[75,30] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/internal/memory/impl/UnsafeUtil.java:[75,45] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/internal/memory/impl/UnsafeUtil.java:[98,44] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/internal/memory/impl/UnsafeUtil.java:[22,15] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/internal/memory/impl/UnsafeUtil.java:[22,15] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/internal/memory/impl/UnsafeUtil.java:[22,15] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/internal/partition/impl/InternalPartitionServiceImpl.java:[431,37] error: cannot find symbol
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/internal/memory/impl/UnsafeUtil.java:[44,24] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/internal/memory/impl/UnsafeUtil.java:[49,8] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/internal/memory/impl/UnsafeUtil.java:[67,19] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/internal/memory/impl/UnsafeUtil.java:[69,19] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/internal/memory/impl/UnsafeUtil.java:[71,70] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/internal/memory/impl/UnsafeUtil.java:[73,23] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/internal/memory/impl/UnsafeUtil.java:[75,30] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/internal/memory/impl/UnsafeUtil.java:[75,45] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/internal/memory/impl/UnsafeUtil.java:[98,44] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
</details>
</pre>

File: hazelcast/src/main/java/com/hazelcast/internal/partition/impl/PartitionStateManager.java (hunk 1 above)
Comment (vbekiaris) [reply hazelcast/src/main/java/com/hazelcast/internal/partition/impl/PartitionStateManager.java]: good catch, `PartitionStateManager` and `InternalPartitionServiceImpl` listened for cluster version changes in order to initialize partition versions on transition from 4.0 to 4.1 cluster version. This is no longer necessary as 4.2 members can only partitipate in 4.1 / 4.2 clusters -> partition version is already initialized before a 4.2 member can join the cluster.

**Review by taburet — APPROVED** [review]

**Review by sancar — APPROVED** [review]

**Comment by vbekiaris** [comment]
run-lab-run

**Pull request merged**